### PR TITLE
Minor changes to support required fields and fix to have Iterable parameters 

### PR DESCRIPTION
**Pull request by guid-empty**
I tried to use this builder ob my project configured on dart 3.0.0/flutter 3.10 and faced with some problems
- it's was not able to generated isolate proxy methods (@SquadronMethod annotated) with named and required not null parameters
- and it's was some mistakes in marshalling with Iterable or List parameters

Please review the fix
Thanks

### CODE REVIEW

**Review by d-markey — APPROVED** [review]

File: lib/src/annotations/squadron_parameter.dart
Comment (d-markey) [inline lib/src/annotations/squadron_parameter.dart:1]: For the required part, maybe the right way to go would be to have `param.isRequiredNamed || param.isRequiredPositional`? Or maybe just `param.isRequired`. The objective here is to make sure Squadron will emit a `required` keyword when generating the code.

Maybe you could post an example of a method definition for which squadron_builder fails generating the right code?

Comment (guid-empty) [reply lib/src/annotations/squadron_parameter.dart:1]: Hi

>> Or maybe just param.isRequired.

I'll try to check this case today maybe or tomorrow. Must likely the problem is in (param.isOptionalNamed || param.isOptionalPositional) part of previous condition to generate "required" part of method declaration

Comment (guid-empty) [reply lib/src/annotations/squadron_parameter.dart:1]: > For the required part, maybe the right way to go would be to have `param.isRequiredNamed || param.isRequiredPositional`? Or maybe just `param.isRequired`. The objective here is to make sure Squadron will emit a `required` keyword when generating the code.
> 
> Maybe you could post an example of a method definition for which squadron_builder fails generating the right code?


Hi @d-markey !
I checked several cases and what I can see.

All these flags  - isRequiredNamed isRequiredPositional isRequiredPositional - are responsible for ability to declare the parameters which should be NotNullable (without "required" keyword but required to have a value) or Nullable but explicitly declared as Nullable with "required" keyword.

```
/// Return `true` if this parameter is either a required positional
/// parameter, or a named parameter with the `required` keyword.
///
/// Note: the presence or absence of the `@required` annotation does not
/// change the meaning of this getter. The parameter `{@required int x}`
/// will return `false` and the parameter `{@required required int x}`
/// will return `true`.
bool get isRequired;

/// Return `true` if this parameter is both a required and named parameter.
/// Named parameters that are annotated with the `@required` annotation are
/// considered optional.  Named parameters that are annotated with the
/// `required` syntax are considered required.
bool get isRequiredNamed;

/// Return `true` if this parameter is both a required and positional
/// parameter.
bool get isRequiredPositional;
```

I checked code generation for my case and there are some results:
Squadron service with two methods declared as:
```
void _handleMessage(telegram.TeleDartMessage message)

Future<dynamic> sendQuestionCardDemoCommand({
  required QuizQuestionModel quizQuestion,
  required IdOfChat participantChatId,
}) async {

```
will be look for param.isRequired like this:
```
@override
void _handleMessage(required TeleDartMessage message) => throw UnimplementedError();
-----------------------^^^^^^^- incorrect behavior


@override
Future<dynamic> sendQuestionCardDemoCommand({required QuizQuestionModel quizQuestion, required IdOfChat participantChatId}) => send(
```

will be look for param.isRequiredPositional like this:

```
@override
Future<dynamic> sendQuestionCardDemoCommand({QuizQuestionModel quizQuestion, IdOfChat participantChatId}) => send
------------------------------^^^^^^^^ - lost "required" keywprd 

```

will be look for param.isRequiredNamed like this - must correct codegeneration:
```
@override
void _handleMessage(telegram.TeleDartMessage message)

@override
Future<dynamic> sendQuestionCardDemoCommand(
        {required QuizQuestionModel quizQuestion,
        required IdOfChat participantChatId}) =>
    send(
```

I'm sure we need to use only isRequiredNamed to generate "required" keyword for parameter declarations when such parameters are declared as required to explicitlty pass any value (nullable/not nullable)

**Comment by guid-empty** [comment]
Hi @d-markey !
Is some problems here to merge the code changes to your project?
Should I just close request? Will be it enough?

**Comment by d-markey** [comment]
Hey @guid-empty im away from computers right now, will look into this as soon as I return. Sorry for the delay but I haven't forgotten

**Pull request merged**